LanguageFeatures/Extension-types/static_analysis_extension_types_A08_t07.dart

```diff
@@ -38,10 +39,13 @@ test1<X extends Future<NumET>>(X x) async {
   }
 }
 
-test2<X extends FutureOr<num>>(X x) async {
+test2<X extends num>(X x) async {
   if (x is int) {
     await x; // No error here, int is compatible with await
   }
+  if (x is FutureOr<IntET>) {
```

Comment:
Can't promote `x` because `FutureOr<IntET>` is not a subtype of `num`.

Reply:
Fixed